src/go/guestagent/pkg/docker/events.go

```diff
@@ -186,7 +186,7 @@ func createPortMapping(ports []types.Port) (nat.PortMap, error) {
 		}
 
 		portBinding := nat.PortBinding{
-			HostIP:   port.IP,
+			HostIP:   containerd.ValidateHostIP(port.IP),
```

Comment:
That seems kind of awkward; is there some shared package we can use to put this code?

Reply:
I thought about this myself too, however, there isn't any suitable package that could accommodate this.